packages/openapi-v3/src/controller-spec.ts

```diff
@@ -200,20 +202,22 @@ function resolveControllerSpec(constructor: Function): ControllerSpec {
 
     for (const p of paramTypes) {
       if (isComplexType(p)) {
-        if (!spec.definitions) {
-          spec.definitions = {};
+        if (!spec.components) {
+          spec.components = {};
+        }
+        if (!spec.components.schemas) {
+          spec.components.schemas = {};
         }
         const jsonSchema = getJsonSchema(p);
         const openapiSchema = jsonToSchemaObject(jsonSchema);
-
         if (openapiSchema.definitions) {
           for (const key in openapiSchema.definitions) {
-            spec.definitions[key] = openapiSchema.definitions[key];
```

Comment:
Make sure with the newest conversion introduced, that nested `components/schemas` don't exist. May need to fiddle around with the test to make sure this is the case.